### ISSUE TITLE
Plugin does not work under SDK Tools Rev. 17

### ISSUE THREAD

**Issue by klu2** [jvoegele/gradle-android-plugin#52]
I've just upgraded to the Android SDK Tools Rev. 17, and have now the following exception when executing any of the tasks in my gradle android project (with rev. 16 everything was fine)
- Exception is:
  org.gradle.api.GradleScriptException: A problem occurred evaluating project ':jawoi-android'.
      at org.gradle.groovy.scripts.internal.DefaultScriptRunnerFactory$ScriptRunnerImpl.run(DefaultScriptRunnerFactory
  .java:54)
      at org.gradle.configuration.DefaultScriptPluginFactory$ScriptPluginImpl.apply(DefaultScriptPluginFactory.java:12
  7)
      at org.gradle.configuration.BuildScriptProcessor.evaluate(BuildScriptProcessor.java:38)
      at org.gradle.configuration.LifecycleProjectEvaluator.evaluate(LifecycleProjectEvaluator.java:43)
      at org.gradle.api.internal.project.AbstractProject.evaluate(AbstractProject.java:462)
      at org.gradle.api.internal.project.AbstractProject.evaluate(AbstractProject.java:74)
      at org.gradle.configuration.ProjectEvaluationConfigurer.execute(ProjectEvaluationConfigurer.java:23)
      at org.gradle.configuration.ProjectEvaluationConfigurer.execute(ProjectEvaluationConfigurer.java:21)
      at org.gradle.configuration.DefaultBuildConfigurer$1.execute(DefaultBuildConfigurer.java:38)
      at org.gradle.configuration.DefaultBuildConfigurer$1.execute(DefaultBuildConfigurer.java:35)
      at org.gradle.api.internal.project.AbstractProject.configure(AbstractProject.java:438)
      at org.gradle.api.internal.project.AbstractProject.allprojects(AbstractProject.java:433)
      at org.gradle.configuration.DefaultBuildConfigurer.configure(DefaultBuildConfigurer.java:35)
      at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:139)
      at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:110)
      at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:78)
      at org.gradle.launcher.cli.RunBuildAction.execute(RunBuildAction.java:42)
      at org.gradle.launcher.cli.RunBuildAction.execute(RunBuildAction.java:28)
      at org.gradle.launcher.exec.ExceptionReportingAction.execute(ExceptionReportingAction.java:32)
      at org.gradle.launcher.exec.ExceptionReportingAction.execute(ExceptionReportingAction.java:21)
      at org.gradle.launcher.cli.CommandLineActionFactory$WithLoggingAction.execute(CommandLineActionFactory.java:302)
  
  ```
  at org.gradle.launcher.cli.CommandLineActionFactory$WithLoggingAction.execute(CommandLineActionFactory.java:286)
  
  at org.gradle.launcher.Main.doAction(Main.java:48)
  at org.gradle.launcher.exec.EntryPoint$1.execute(EntryPoint.java:53)
  at org.gradle.launcher.exec.EntryPoint$1.execute(EntryPoint.java:51)
  at org.gradle.launcher.exec.Execution.execute(Execution.java:28)
  at org.gradle.launcher.exec.EntryPoint.run(EntryPoint.java:39)
  at org.gradle.launcher.Main.main(Main.java:39)
  at org.gradle.launcher.ProcessBootstrap.runNoExit(ProcessBootstrap.java:51)
  at org.gradle.launcher.ProcessBootstrap.run(ProcessBootstrap.java:33)
  at org.gradle.launcher.GradleMain.main(GradleMain.java:24)
  at org.gradle.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:33)
  at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:130)
  at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:47)
  ```
  
  Caused by: : java.lang.NullPointerException
      at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:116)
      at org.gradle.api.internal.project.ant.BasicAntBuilder.nodeCompleted(BasicAntBuilder.java:71)
      at org.gradle.api.internal.project.ant.BasicAntBuilder.doInvokeMethod(BasicAntBuilder.java:86)
      at org.gradle.api.internal.project.DefaultAntBuilder.super$3$invokeMethod(DefaultAntBuilder.groovy)
      at org.gradle.api.internal.project.DefaultAntBuilder.invokeMethod(DefaultAntBuilder.groovy:37)
      at com.jvoegele.gradle.plugins.android.AndroidSetup_r14.setup(AndroidSetup_r14.groovy:49)
      at com.jvoegele.gradle.plugins.android.AndroidSetup$setup.call(Unknown Source)
      at com.jvoegele.gradle.plugins.android.AndroidPlugin.androidSetup(AndroidPlugin.groovy:89)
      at com.jvoegele.gradle.plugins.android.AndroidPlugin.apply(AndroidPlugin.groovy:77)
      at com.jvoegele.gradle.plugins.android.AndroidPlugin.apply(AndroidPlugin.groovy)
      at org.gradle.api.internal.plugins.DefaultProjectsPluginContainer.providePlugin(DefaultProjectsPluginContainer.j
  ava:107)
      at org.gradle.api.internal.plugins.DefaultProjectsPluginContainer.addPluginInternal(DefaultProjectsPluginContain
  er.java:71)
      at org.gradle.api.internal.plugins.DefaultProjectsPluginContainer.apply(DefaultProjectsPluginContainer.java:37)
      at org.gradle.api.internal.plugins.DefaultObjectConfigurationAction.applyPlugin(DefaultObjectConfigurationAction
  .java:101)
      at org.gradle.api.internal.plugins.DefaultObjectConfigurationAction.access$200(DefaultObjectConfigurationAction.
  java:32)
      at org.gradle.api.internal.plugins.DefaultObjectConfigurationAction$3.run(DefaultObjectConfigurationAction.java:
  72)
      at org.gradle.api.internal.plugins.DefaultObjectConfigurationAction.execute(DefaultObjectConfigurationAction.jav
  a:114)
      at org.gradle.api.internal.project.AbstractProject.apply(AbstractProject.java:840)
      at org.gradle.api.Project$apply.call(Unknown Source)
      at org.gradle.api.internal.project.ProjectScript.apply(ProjectScript.groovy:34)
      at org.gradle.api.Script$apply.callCurrent(Unknown Source)
      at build_5rarmlkctuq80rmcosn8cs5ams.run(C:\catalysts\workspaces\jawoitv\jawoi\jawoi-android\build.gradle:28)
      at org.gradle.groovy.scripts.internal.DefaultScriptRunnerFactory$ScriptRunnerImpl.run(DefaultScriptRunnerFactory
  .java:52)
      ... 33 more
  Caused by: java.lang.NullPointerException
      at com.android.ant.NewSetupTask.sanitizePaths(NewSetupTask.java:745)
      at com.android.ant.NewSetupTask.processReferencedLibraries(NewSetupTask.java:573)
      at com.android.ant.NewSetupTask.execute(NewSetupTask.java:300)
      at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:291)
      at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
      ... 55 more

**Comment by klu2** [comment]
unfortunately I cannot find the source code of NewSetupTask.java anywhere, therefore I even can't debug into it...

**Comment by klu2** [comment]
Anyways, when exchanging the tools/lib/anttask.jar with the revision 16, it works. The tool notes mention that ANT 1.8 is required now, maybe that is the point: http://developer.android.com/sdk/tools-notes.html

**Comment by shawnlauzon** [comment]
I can confirm, I'm getting the same error. NewSetupTask should be part of the Android source, let me see if I can dig it up.

**Comment by shawnlauzon** [comment]
Here's the line causing the NPE:        File outputFile = new File(antProject.getProperty("out.absolute.dir"));

**Comment by ealden** [comment]
Fixed in #55
